openselfsup/models/necks.py

```diff
@@ -230,6 +230,89 @@ def forward(self, x):
         return [self.mlp(x.view(x.size(0), -1))]
 
 
+@NECKS.register_module
+class NonLinearNeckBYOL(nn.Module):
```

Comment:
There is a lot of repeating codes between NonLinearNeckBYOL and NonLinearNeckSimCLR, please merge them into one class.